draft-ietf-dnsop-domain-verification-techniques.md

```diff
@@ -170,13 +187,15 @@ The RECOMMENDED format is application-specific underscore prefix labels. Domain
 
 ### Random Token {#random-token}
 
-A unique token used in the challenge. It should be a random value with the following properties:
+A unique token used in the challenge. It should be a random value issued to the user by the provider with the following properties:
 
 1. MUST have at least 128 bits of entropy.
-2. base64url ({{!RFC4648, Section 5}}) encoded.
+2. base64url ({{!RFC4648, Section 5}}) encoded or base16 ({{!RFC4648, Section 8}}) encoded.
 
 See {{RFC4086}} for additional information on randomness requirements.
 
+Hexadecimal base16 encoding is recommended when the random token would exist in a DNS label such as in a CNAME target.
```

Comment:
```suggestion
Hexadecimal base16 encoding is RECOMMENDED when the random token would exist in a DNS label such as in a CNAME target.  This is because base64 relies mixed case (and DNS is case-insensitive as clarified in {{RFC4343}}) and because some base64 characters ("/", "+", and "=") may not be permitted by implementations that limit allowed characters to those allowed in hostnames.
```

How is this?  We'd also need to add RFC4343 to the informational references.

Reply:
sounds good!